lib/templates/src/components/Chat.vue

```diff
@@ -0,0 +1,42 @@
+<template>
+  <div>
+    <div id="chat-box">
+      <p v-for="message in messages" :key="message" v-text="message"></p>
+    </div>
+    <div class="magpie-view-answer-container">
+      <textarea
+        ref="text"
+        cols="50"
+        placeholder="Type your message to the other participant here."
+      ></textarea>
+      <button @click.stop="send()">
+        Send
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+const EVENT_CHAT_MESSAGE = 'chat_message';
+
+export default {
+  name: 'Chat',
+  data() {
+    return {
+      messages: []
+    };
+  },
+  socket: {
+    [EVENT_CHAT_MESSAGE](payload) {
+      this.messages.push(payload);
```

Comment:
> I guess you haven't added the listener to actually append the message, when a new message arrives?

This is the listener. :)

Reply:
`socket` is a new Vue option introduced by the magpie plugin below, allowing registration of socket event listeners.

Reply:
I see 😄 . Somehow I overlooked the `this.messages.push(payload)` part. This seems like a really concise implementation and I like it 👍